sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/planning/patterns.scala

```diff
@@ -91,6 +97,27 @@ object PhysicalOperation extends PredicateHelper {
           .map(Alias(_, a.name)(a.exprId, a.qualifier)).getOrElse(a)
     }
   }
+
+  private def getDeterminedExpression(expr: Expression): Option[Expression] = {
```

Comment:
Also some comments about this function at the top with small snippets of input and output.